spec/providers/atmosphere/optimization_strategy/default_spec.rb

```diff
@@ -37,21 +37,27 @@ def appliance(options)
     )
   end
 
-  let!(:wf) { create(:workflow_appliance_set) }
-  let!(:wf2) { create(:workflow_appliance_set) }
-  let!(:openstack) { create(:openstack_with_flavors, funds: [fund]) }
-  let!(:fund) { create(:fund) }
-  let!(:shareable_appl_type) { create(:shareable_appliance_type) }
-  let!(:tmpl_of_shareable_at) { create(:virtual_machine_template, appliance_type: shareable_appl_type, compute_site: openstack)}
-
   context 'new appliance created' do
+    let(:wf) { create(:workflow_appliance_set) }
+    let(:openstack) { create(:openstack_with_flavors, funds: [fund]) }
+    let(:fund) { create(:fund) }
+    let(:shareable_appl_type) { create(:shareable_appliance_type) }
+
+    before do
+      # os family reused by AT, Flavors - simply magic :(
```

Comment:
It's not magic; `openstack_with_funds` is an OpenStack-type CS with 5 flavors created automatically.